### PR TITLE
LPS-159061 Upgrade Process to External Reference Code Field

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPS-159061

### CODE REVIEW

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:reevaluate:1377401_4029

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v3_20_0/ObjectFieldUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+										"external-reference-code"));
+							}
+						},
+						"Label"));
+
+				preparedStatement2.setString(21, "externalReferenceCode");
+				preparedStatement2.setNull(22, 0);
+				preparedStatement2.setInt(23, 0);
+				preparedStatement2.setInt(24, 0);
+				preparedStatement2.setInt(25, 1);
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v3_20_0/ObjectFieldUpgradeProcess.java:130]: Hi @p-albuquerque,

system_, state_, required and other boolean fields should be defined as boolean, you must use:

`preparedStatement2.setBoolean`

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v3_20_0/ObjectFieldUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+				preparedStatement2.addBatch();
+
+				preparedStatement2.executeBatch();
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v3_20_0/ObjectFieldUpgradeProcess.java:134]: Put the batch execution outside the loop

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:forward

**Comment by seltong** [comment]
Resent [here](https://github.com/liferay-objects/liferay-portal/pull/990)

**Pull request closed without merging**